components/script/dom/event.rs

```diff
@@ -482,7 +482,14 @@ fn invoke(window: Option<&Window>,
     event.current_target.set(Some(object));
 
     // Step 5.
-    inner_invoke(window, object, event, &listeners);
+    if inner_invoke(window, object, event, &listeners) {
+        // <https://html.spec.whatwg.org/multipage/#unload-a-document>
+        // Step 9.
+        // Required to establish the 'salvageable' state of document as part of unloading.
+        if event.canceled.get() != EventDefault::Prevented {
```

Comment:
I don't think this is the right place to do this. See comment below.

Reply:
Now that I've looked into this more, this doesn't make much sense since we are checking if  `canceled` is `Prevented` and if so, we change it to `Handled`. This shouldn't have an effect since in the `unload` steps you just check if cancelled is not equal to `Allowed`.

Reply:
Yes that makes sense, and the problem is that the default is 'Allowed', to which it is set whether there was a listener handling the event or not(so long as the listener didn't preventDefault), hence the need to specifically mark it as 'Handled' to know the difference between 'no listener triggered', and 'a listener was triggered, but the default was not prevented(and the document is not salvageable)'.